### PR TITLE
Dbodor/testing slides

### PR DESCRIPTION
**Pull request by DaniBodor**
updates to testing slides:

- added images and animation to many of the slide
- rephrased and edited logic of some slides
- added a slide on how test metrics can be "misleading"
- I integrated pytest into the flow a bit more and stuck with a temperature conversion example (same example as in modularization episode)
- removed the "pure functions recap", as I think it's not really part of this module.
  - not sure how we feel about this. I find it distracts a bit from the testing, but it is not really addressed in this much detail anywhere else. We could decide to emphasize it more in the modularization episode, but find that one is quite dense already, so don't know if I'd want to put more info about it there either.
- streamlined fizzbuzz/TDD part a bit

### CODE REVIEW

File: testing/00_title.md [hunk 1]
```diff
@@ -44,162 +73,235 @@ plugins: RevealMarkdown, RevealChalkboard, RevealHighlight, RevealMath.KaTeX, Re
 ## Test Types
 
 <ul>
-  <li>Unit test
-  <ul class="fragment fade-up" data-fragment-index="1">
-    <li>Smallest possible unit</li>
+  <li class="fragment fade-up">Exceptions in the code base
```
Comment (lyashevska) [inline testing/00_title.md:76]: Maybe make Unit test bold to make it clear that we only will focus on them? Perhaps  put them on the top of the list.

File: testing/00_title.md [hunk 2]
```diff
@@ -44,162 +73,235 @@ plugins: RevealMarkdown, RevealChalkboard, RevealHighlight, RevealMath.KaTeX, Re
...
+  <ul>
+    <li>Intended to handle "expected" problems</li>
+    <li>Sound an alarm as soon as the problem arises</li>
+    <li>Provide clear feedback to the user</li>
+  </ul></li>
+  <li class="fragment fade-up">Unit testing
```
Comment (lyashevska) [inline testing/00_title.md]: Maybe highlight Unit testing to make it clear that we only will focus on them? Perhaps put them on the top of the list.

File: testing/00_title.md [hunk 3]
```diff
@@ -44,162 +73,235 @@ plugins: RevealMarkdown, RevealChalkboard, RevealHighlight, RevealMath.KaTeX, Re
...
-</code></pre>
-
-</div>
-
 ---
 
 <!-- .slide: data-state="standard" data-background="./files/whitebg.png" -->
 
-## Test!
+## Using pytest to run tests in Python
```
Comment (lyashevska) [inline testing/00_title.md]:
```suggestion
## Using pytest to run Unit tests in Python
```

File: testing/00_title.md [hunk 4]
```diff
@@ -44,162 +73,235 @@ plugins: RevealMarkdown, RevealChalkboard, RevealHighlight, RevealMath.KaTeX, Re
...
 
-<pre><code style="overflow: hidden;" data-trim class="bash" data-line-numbers="1|1-9">
-$ pytest example.py
+[Pytest](https://docs.pytest.org/en/7.3.x/) is the recommended python testing framework
```
Comment (lyashevska) [inline testing/00_title.md]: Maybe mention other frameworks, e.g. Junit in java, etc. to make it language-agnostic.

File: testing/00_title.md [hunk 5]
```diff
@@ -44,162 +73,235 @@ plugins: RevealMarkdown, RevealChalkboard, RevealHighlight, RevealMath.KaTeX, Re
...
 
+The `pytest` command will run all functions starting with "test_" from all files starting with "test_".
+<!-- .element: class="fragment" -->
+
 ---
 
 <!-- .slide: data-state="standard" data-background="./files/whitebg.png" -->
 
-## Breaking Things
+## Pytest Excercise
```
Comment (lyashevska) [inline testing/00_title.md]:
```suggestion
## Pytest Exercise
```

File: testing/00_title.md [hunk 6]
```diff
@@ -216,53 +318,19 @@ def test_paint_cows():
...
 ## Take-away
 
 - Use pure functions when possible 👌
+  - Do you remember what these are? 💭
 - Testing does not have to be hard 👏
-  - You test anyways, but then throw the test away 🧐
+  - You often test anyway, but then throw the test away 🧐
+  - Use pytest if programming with Python 🎭
 - You don't have to strive for 💯% test coverage
+  - But be smart about what you ar testing 🧠
```
Comment (lyashevska) [inline testing/00_title.md]:
```suggestion
  - But be smart about what you are testing 🧠
```

**Review by lyashevska — CHANGES_REQUESTED** [review]
Great improvements @DaniBodor. Thank you. Just a few comments, please merge it when you are ready.

File: testing/00_title.md (hunk 2 above)
Comment (DaniBodor) [reply testing/00_title.md]: Good point. I will find a way to emphasize that this is what we will focus on. I still like the order the way it is (I organized it more or less small-large scale/simple-complex, which I think makes sense), but will think of some other way to emphasize.

I added an animation to remove the two other headers. Ideally I only wanted to fade them out, but I didnt find out how :)

File: testing/00_title.md (hunk 3 above)
Comment (DaniBodor) [reply testing/00_title.md]: I prefer the title shorter, and in principle running integration tests would still work the same way.

**Review by lyashevska — APPROVED** [review]
Looking good!

**Review by lyashevska — APPROVED** [review]
Looking good!

**Pull request merged**